File: Backend/generic-free-spots-backend/src/main/java/be/PXLResearch/code4belgium/general/Filter.java
package be.PXLResearch.code4belgium.general;

import be.PXLResearch.code4belgium.enums.DataType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "filters", indexes = {
        @Index(name = "idx_filter_name", columnList = "name"),
        @Index(name = "idx_filter_datatype", columnList = "datatype"),
        @Index(name = "idx_filter_value", columnList = "value"),
        @Index(name = "idx_filter_active", columnList = "active")
})
public class Filter {
    @Id
    @GeneratedValue
    private Long id;
    private String name;
    private DataType datatype;
    private String value;
    @ManyToMany(mappedBy = "filters")
    private List<Sector> sectors;
    private boolean active;
}

File: Frontend/gvpp/src/models/filter.ts
export class Filter {
  id: number;
  name: string;
  datatype: string;
  value: string;
  active: boolean;
  inputType: string;

  constructor(id: number, name: string, datatype: string, value: string, active: boolean, inputType: string) {
    this.id = id;
    this.name = name;
    this.datatype = datatype;
    this.value = value;
    this.active = active;
    this.inputType = inputType;
  }
}
